python/lsst/analysis/tools/atools/diffMatched.py

```diff
@@ -115,86 +134,260 @@ def setDefaults(self):
+
+
+class InjectedClassSelector(InjectedObjectSelector):
+    """A selector for injected objects of a given class."""
+
+    key_class = Field[str](
+        doc="Key for the field indicating the class of the object",
+        default="ref_source_type",
+    )
+    name_class = Field[str](
+        doc="Name of the class of objects",
+    )
+    value_compare = Field[str](
+        doc="Value of the type_key field for objects that are stars",
+        default="DeltaFunction",
+    )
+    value_is_equal = Field[bool](
+        doc="Whether the value must equal value_compare to be of this class",
+        default=True,
+    )
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        result = super().__call__(data, **kwargs)
+        values = data[self.key_class]
+        result &= (values == self.value_compare) if self.value_is_equal else (values != self.value_compare)
+        if self.plotLabelKey:
+            self._addValueToPlotInfo(f"injected {self.name_class}", **kwargs)
+        return result
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield from super().getInputSchema()
+        yield self.key_class, Vector
+
+
+class InjectedStarSelector(InjectedClassSelector):
+    """A selector for injected stars of a given class."""
+
+    def setDefaults(self):
+        self.name_class = "star"
+
+
+class InjectedGalaxySelector(InjectedClassSelector):
+    def setDefaults(self):
+        self.name_class = "galaxy"
+        self.value_is_equal = False
+
+
+class MatchedClassAction(VectorAction):
+    """Action to return whether matched coadd objects are the same class."""
+
+    key_is_ref_galaxy = pexConfig.Field[str](
+        doc="The key of the boolean field selecting reference galaxies",
+    )
+    key_is_ref_star = pexConfig.Field[str](
+        doc="The key of the boolean field selecting reference stars",
+    )
+    key_is_target_galaxy = pexConfig.Field[str](
+        doc="The key of the boolean field selecting observed galaxies",
+    )
+    key_is_target_star = pexConfig.Field[str](
+        doc="The key of the boolean field selecting observed starss",
+    )
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        mask = kwargs.get("mask")
+        is_ref_galaxy = data[self.key_is_ref_galaxy]
+        is_ref_star = data[self.key_is_ref_star]
+        is_target_galaxy = data[self.key_is_target_galaxy]
+        is_target_star = data[self.key_is_target_star]
+        if mask:
+            is_ref_galaxy = is_ref_galaxy[mask]
+            is_ref_star = is_ref_star[mask]
+            is_target_galaxy = is_target_galaxy[mask]
+            is_target_star = is_target_star[mask]
+        return (is_ref_galaxy & is_target_galaxy) | (is_ref_star & is_target_star)
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield self.key_is_ref_galaxy, Vector
+        yield self.key_is_ref_star, Vector
+        yield self.key_is_target_galaxy, Vector
+        yield self.key_is_target_star, Vector
+
```

Comment:
All of these actions should be defined in the appropriate actions file. Or make a new one for injected stuff but other people may want to use these things and they should be in the actions folder where people will find them.

Reply:
`actions/vector/selectors.py` is starting to get rather long. Any name suggestions? `injectedSelectors.py` is more readable but `selectorsInjected.py` sorts better.

Reply:
Both are readable to me, but maybe I'm biased by familiarity.